modifiers/string.go

```diff
@@ -99,6 +100,37 @@ func TitleCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param
 	return nil
 }
 
+var namePatterns = []map[string]string{
+	{`[^\pL-\s']`: ""}, // cut off everything except [ alpha, hyphen, whitespace, apostrophe]
+	{`\s{2,}`: " "},    // trim more than two whitespaces to one
+	{`-{2,}`: "-"},     // trim more than two hyphens to one
+	{`'{2,}`: "'"},     // trim more than two apostrophes to one
+	{`( )*-( )*`: "-"}, // trim enclosing whitespaces around hyphen
+}
+
+// NameCase Trims, strips numbers and special characters (except dashes and spaces separating names),
+// converts multiple spaces and dashes to single characters, title cases multiple names.
+// Example: "3493€848Jo-$%£@Ann " -> "Jo-Ann", " ~~ The Dude ~~" -> "The Dude", "**susan**" -> "Susan",
+// " hugh fearnley-whittingstall" -> "Hugh Fearnley-Whittingstall"
+func NameCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param string) error {
+	s, ok := v.Interface().(string)
+	if !ok {
+		return nil
+	}
+	first := onlyOne(strings.ToLower(s), namePatterns)
+	v.SetString(strings.Title(regexp.MustCompile(`[\p{L}]([\p{L}|[:space:]\-']*[\p{L}])*`).FindString(first)))
```

Comment:
Fi the regex can be made into a global that would be more efficient.

Reply:
Ok. I'll do it, but now I'm asking if there is a better way to do this, maybe without regex? What do you think?

Reply:
Done.

Reply:
I think it'd be more efficient without the regex, but regex works for the first iteration and con be optimized later.